packages/plugin-core/src/WSUtilsV2.ts

```diff
@@ -268,4 +271,78 @@ export class WSUtilsV2 implements IWSUtilsV2 {
     const fnameWithExtension = `${fname}.md`;
     return this.openFileInEditorUsingFullFname(vault, fnameWithExtension);
   }
+
+  async findDuplicateNoteFromDocument(document: vscode.TextDocument) {
+    const fsPath = document.uri.fsPath;
+    const { vaults, wsRoot, engine } = this.extension.getDWorkspace();
+    let vault;
+    try {
+      vault = VaultUtils.getVaultByFilePath({
+        vaults,
+        wsRoot,
+        fsPath,
+      });
+    } catch (error: any) {
+      // document doesn't exist in workspace.
+      return;
+    }
+
+    // we do this because the note in document would _not_ be in our store
+    // if it is a duplicate note.
+    const currentNote = file2Note(fsPath, vault);
```

Comment:
Looking at the code - this may throw if we're not in a Dendron note at all?  Say someone opens a random non-Dendron Markdown file with no frontmatter, or a plain text file.  Are we still good here?

Reply:
`file2Note` will result in a `NoteProps` with the frontmatter newly populated, in which case the logic moving forward will detect that there is no note that has the same id in store, which short circuit the rest of the checks.

So we are still good here. Unless you think specifically checking for that case is preferable.

Reply:
Anyways, I'll add a test for the case you are describing.